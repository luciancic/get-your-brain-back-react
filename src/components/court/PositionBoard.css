.position-board {
    border-radius: 20px;
    display: grid;
    grid-gap: 1rem;
    grid-template-columns: 1fr 1fr 1fr;
    grid-template-rows: 1fr 1fr 1fr;
    justify-items: center;
    align-items: center;
    padding: 1rem;
}

.position-circle {
    transition: background-color 200ms ease-in-out;
    height: 70px;
    width: 70px;
    border-radius: 100%;
}

.position-buttons button {
    width: 11em;
}

@media (max-height: 500px) and (min-aspect-ratio: 16/10) {
    .position-circle {
        height: 45px;
        width: 45px;
    }
}