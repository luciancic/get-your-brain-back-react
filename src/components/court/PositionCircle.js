import React from 'react';

export default (props) => <div className={`position-circle green ${props.active ? "" : "lighten-4"}`} />