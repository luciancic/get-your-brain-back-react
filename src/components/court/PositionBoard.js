import React from 'react'
import PositionCircle from './PositionCircle'

function PositionBoard(props) {
    const { active, currentPosition } = props
    const arr = [ 1, 2, 3, 4, 5, 6, 7, 8, 9 ]

    return <div className="court__board" >
        {arr.map((i) => {
            return <PositionCircle active={ active && i === currentPosition} key={i} />
        })}
    </div>
}

export default PositionBoard
