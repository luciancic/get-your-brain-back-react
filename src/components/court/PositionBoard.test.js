import React from 'react';
import { shallow } from 'enzyme';
import PositionBoard from './PositionBoard';
import PositionCircle from './PositionCircle';

it('renders PositionBoard', () => {
  const wrapper = shallow(<PositionBoard active={true} currentPosition={2} />);
  const circles = wrapper.find(PositionCircle);

  expect( wrapper ).toMatchSnapshot();
  expect( circles ).toHaveLength(9);
  expect( wrapper.find({ active: false }) ).toHaveLength(8);
  expect( circles.at(1).prop('active') ).toBeTruthy();
});