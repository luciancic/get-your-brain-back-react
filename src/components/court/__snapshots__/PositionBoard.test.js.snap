// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PositionBoard 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <_default
    active={true}
    currentPosition={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <_default
          active={false}
        />,
        <_default
          active={true}
        />,
        <_default
          active={false}
        />,
        <_default
          active={false}
        />,
        <_default
          active={false}
        />,
        <_default
          active={false}
        />,
        <_default
          active={false}
        />,
        <_default
          active={false}
        />,
        <_default
          active={false}
        />,
      ],
      "className": "teal lighten-5 position-board",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "active": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "active": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "active": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "active": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "function",
        "props": Object {
          "active": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "function",
        "props": Object {
          "active": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "function",
        "props": Object {
          "active": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "8",
        "nodeType": "function",
        "props": Object {
          "active": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "9",
        "nodeType": "function",
        "props": Object {
          "active": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <_default
            active={false}
          />,
          <_default
            active={true}
          />,
          <_default
            active={false}
          />,
          <_default
            active={false}
          />,
          <_default
            active={false}
          />,
          <_default
            active={false}
          />,
          <_default
            active={false}
          />,
          <_default
            active={false}
          />,
          <_default
            active={false}
          />,
        ],
        "className": "teal lighten-5 position-board",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "active": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "active": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "active": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "active": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "active": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "active": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "function",
          "props": Object {
            "active": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "function",
          "props": Object {
            "active": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "function",
          "props": Object {
            "active": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
